src/test/java/org/broadinstitute/hellbender/utils/read/AlignmentUtilsUnitTest.java

```diff
@@ -848,6 +848,172 @@ public void testLeftAlignIndelData(final Cigar originalCigar, final Cigar expect
         Assert.assertTrue(expectedCigar.equals(actualCigar), "Wrong left alignment detected for cigar " + originalCigar.toString() + " to " + actualCigar.toString() + " but expected " + expectedCigar.toString() + " with repeat length " + repeatLength);
     }
 
+    @DataProvider(name = "LeftAlignIndelWithLimitDataProvider")
+    public Object[][] makeLeftAlignIndelWithLimitDataProvider() {
+        List<Object[]> tests = new ArrayList<>();
+
+        final byte[] repeat1Reference = "ABCDEFGHIJKLMNOPXXXXXXXXXXABCDEFGHIJKLMNOP".getBytes();
+        final byte[] repeat2Reference = "ABCDEFGHIJKLMNOPXYXYXYXYXYABCDEFGHIJKLMNOP".getBytes();
+        final byte[] repeat3Reference = "ABCDEFGHIJKLMNOPXYZXYZXYZXYZABCDEFGHIJKLMN".getBytes();
+        final int referenceLength = repeat1Reference.length;
+
+        for (int indelStart = 0; indelStart < repeat1Reference.length; indelStart++) {
+            for (final int indelSize : Arrays.asList(0, 1, 2, 3, 4)) {
+                for (final char indelOp : Arrays.asList('D', 'I')) {
+                    for (int leftmostAllowedAlignment = 0; leftmostAllowedAlignment < indelStart; leftmostAllowedAlignment++) {
+                        if (indelOp == 'D' && indelStart + indelSize >= repeat1Reference.length)
+                            continue;
+
+                        final int readLength = referenceLength - (indelOp == 'D' ? indelSize : -indelSize);
+
+                        // create the original CIGAR string
+                        final GATKRead read = ArtificialReadUtils.createArtificialRead(header, "myRead", 0, 1, readLength);
```

Comment:
I think this is a good thorough test that helps ensure that this method is ding what it purports to. I do still think there is a case that i'm not strictly convinced this method is handling correctly (or at least expectedly) according to your usage of it in the tool. All of these test cases involve a read spanning the entirety of the contig, which means when we left align the indel we end up with the indel sitting right in the middle of the read (though surely correctly for various types of repetitions). The issue is that I don't see any test for the behavior of this method if we expect the indel to fall off of the edge of the read. I would like to see some assertion that if we start the read in the middle of say a 3base str that it will necessarily fall off the left end of the read if we run it, regardless of where the read/indel starts modulo 3 relative to the start of the indel. If we are sure according to a test that the indel will fall off of the read then we can continue to use your auto-expansion code.

Reply:
I have added a test of this to ``AlignmentUtilsUnitTest``.